eval/src/vespa/eval/tensor/dense/dense_tensor_view.cpp

```diff
@@ -273,4 +272,12 @@ DenseTensorView::reduce(join_fun_t op, const std::vector<vespalib::string> &dime
     return dense::reduce(*this, dims, op);
 }
 
+Tensor::UP
+DenseTensorView::reduce(join_fun_t op, const std::vector<vespalib::string> &dimensions) const
+{
+    return dimensions.empty()
+            ? reduce_all(op, _typeRef.dimension_names())
+            : reduce_all(op, dimensions);
```

Comment:
this patch should not call reduce_all, since it should reduce the named dimensions, not all of them...